rdmo/tasks/managers.py

```diff
@@ -10,16 +10,27 @@
 )
 
 
-class TaskQuestionSet(CurrentSiteQuerySetMixin, GroupsQuerySetMixin, AvailabilityQuerySetMixin, models.QuerySet):
+class TaskQuestionSet(CurrentSiteQuerySetMixin, GroupsQuerySetMixin, AvailabilityQuerySetMixin, QuerySet):
 
     def filter_catalog(self, catalog):
-        return self.filter(models.Q(catalogs=None) | models.Q(catalogs=catalog))
+        return self.filter(Q(catalogs=None) | Q(catalogs=catalog))
 
+    def filter_available_tasks_for_project(self, project):
+        site_filter = Q(sites=project.site) | Q(sites__isnull=True)
+        catalogs_filter = Q(catalogs=project.catalog) | Q(catalogs__isnull=True)
+        groups_filter = Q(groups__in=project.groups) | Q(groups__isnull=True)
+        availability_filter = Q(available=True)
```

Comment:
I think you removed the check for the managers and admins here, this was done by `.filter_availability(self.request.user)` and needs the user. This comes from `AvailabilityQuerySetMixin`. I think you also reimplemented the other thinks in `core/managers.py`.

Reply:
Maybe the availability part should stay in `AvailabilityQuerySetMixin` and we need only `filter_for_project` and use it like this 

```python
tasks = Task.objects.filter_for_project(project).filter_availability(self.request.user)
```

alternatively 3 methods

```
        tasks = Task.objects.filter_for_project_site(project) \
                            .filter_for_project_catalog(project) \
                            .filter_for_project_group(project) \
                            .filter_availability(self.request.user)
```

Reply:
I like "my" style better, but I think this is fine (at least no `\`). I would indent the two '.filter...' lines by 4.

Reply:
I've used one-liner queries when there is only a single filter and it fits on one line, for multiple filters I've used this multi-line parentheses notation with the extra indent. So basically best of both worlds ;)